packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt

```diff
@@ -5,71 +5,74 @@ import io.flutter.plugin.common.MethodChannel
 import java.io.IOException
 
 /** Handles the method calls for the plugin.  */
-internal class MethodCallHandler(private val share: Share, private val manager: ShareSuccessManager) : MethodChannel.MethodCallHandler {
+internal class MethodCallHandler(
+    private val share: Share,
+    private val manager: ShareSuccessManager
+) : MethodChannel.MethodCallHandler {
 
-  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
-    when (call.method) {
-      "share" -> {
-        expectMapArguments(call)
-        // Android does not support showing the share sheet at a particular point on screen.
-        share.share(
-          call.argument<Any>("text") as String,
-          call.argument<Any>("subject") as String?,
-          false,
-        )
-        result.success(null)
-      }
-      "shareFiles" -> {
-        expectMapArguments(call)
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            "share" -> {
+                expectMapArguments(call)
+                // Android does not support showing the share sheet at a particular point on screen.
+                share.share(
+                    call.argument<Any>("text") as String,
+                    call.argument<Any>("subject") as String?,
+                    false,
+                )
+                result.success(null)
+            }
+            "shareFiles" -> {
+                expectMapArguments(call)
 
-        // Android does not support showing the share sheet at a particular point on screen.
-        try {
-          share.shareFiles(
-            call.argument<List<String>>("paths")!!,
-            call.argument<List<String>?>("mimeTypes"),
-            call.argument<String?>("text"),
-            call.argument<String?>("subject"),
-            false,
-          )
-          result.success(null)
-        } catch (e: IOException) {
-          result.error(e.message, null, null)
-        }
-      }
-      "shareWithResult" -> {
-        expectMapArguments(call)
-        if(!manager.setCallback(result)) return
+                // Android does not support showing the share sheet at a particular point on screen.
+                try {
+                    share.shareFiles(
+                        call.argument<List<String>>("paths")!!,
+                        call.argument<List<String>?>("mimeTypes"),
+                        call.argument<String?>("text"),
+                        call.argument<String?>("subject"),
+                        false,
+                    )
+                    result.success(null)
+                } catch (e: IOException) {
+                    result.error("Share failed", e.message, null)
```

Comment:
I was considering submitting a similar fix, but keep in mind that this may be a breaking change for some users that rely on the returned error code or error message. At the very least it should coincide with more than a simple PATCH version bump.

Reply:
I doubt that somebody relied on this as otherwise this person would open an issue that `e.message` was set on a place of `errorCode` parameter.

Reply:
Right, I would also doubt it, but no reason not to err on the side of caution in terms of versioning.